src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -1042,4 +1061,492 @@ public void deleteHarvestedDataset(Dataset dataset, DataverseRequest request, Lo
             hdLogger.warning("Failed to destroy the dataset");
         } 
     }
+
+
+
+
+    @Asynchronous
+    public void globusUpload(String jsonData, ApiToken token, Dataset dataset, String httpRequestUrl, User authUser) throws ExecutionException, InterruptedException, MalformedURLException {
+
+        Integer countAll = 0;
+        Integer countSuccess = 0;
+        Integer countError = 0;
+        String logTimestamp = logFormatter.format(new Date());
+        Logger globusLogger = Logger.getLogger("edu.harvard.iq.dataverse.upload.client.DatasetServiceBean." + "GlobusUpload" + logTimestamp);
+        String logFileName = "../logs" + File.separator + "globusUpload_id_" + dataset.getId() + "_" + logTimestamp + ".log";
+        FileHandler fileHandler;
+        boolean fileHandlerSuceeded;
+        try {
+            fileHandler = new FileHandler(logFileName);
+            globusLogger.setUseParentHandlers(false);
+            fileHandlerSuceeded = true;
+        } catch (IOException | SecurityException ex) {
+            Logger.getLogger(DatasetServiceBean.class.getName()).log(Level.SEVERE, null, ex);
+            return;
+        }
+
+        if (fileHandlerSuceeded) {
+            globusLogger.addHandler(fileHandler);
+        } else {
+            globusLogger = logger;
+        }
+
+        globusLogger.info("Starting an globusUpload ");
+
+        String datasetIdentifier = dataset.getStorageIdentifier();
+
+        String storageType = datasetIdentifier.substring(0, datasetIdentifier.indexOf("://") + 3);
+        datasetIdentifier = datasetIdentifier.substring(datasetIdentifier.indexOf("://") + 3);
+
+
+        Thread.sleep(5000);
+
+        JsonObject jsonObject = null;
+        try (StringReader rdr = new StringReader(jsonData)) {
+            jsonObject = Json.createReader(rdr).readObject();
```

Comment:
It might be better to do at least some minimal checks in the API call itself (before you go asynchronous and wait 5 seconds) so you can, for example, return 400/Bad Request if unparsable json is submitted.